aten/src/ATen/cuda/CachingHostAllocator.cpp

```diff
@@ -98,7 +98,8 @@ struct CUDACachingHostAllocatorImpl
             pinned_use_cuda_host_register()) {
       void* ptr = block->ptr_;
       AT_CUDA_CHECK(cudaHostUnregister(ptr));
-      free(ptr);
+      // call C-library free instead of ours (that frees a block)
+      ::free(ptr);
```

Comment:
Is `std::free` more explicit here or is that incorrect?

Reply:
YEah std::free should work too (and we should replace corresponding malloc with std::malloc then)